src/outputs/manifest_output.py

```diff
@@ -0,0 +1,186 @@
+import json
+import os
+from datetime import datetime
+
+
+class ManifestError(Exception):
+    pass
+
+
+class Manifest:
+    """
+    An outgoing file transfer manifest. Accumulates file metadata before
+        writing a JSON manifest file.
+
+    Used by Apache NiFi for data integrity checks before initiating data
+        transfers.
+
+    Attributes
+    ==========
+    outgoing_directory
+        location to write files to
+    pipeline_run_datetime
+        datetime of the current pipeline run, used to version outputs
+    dry_run
+        when True, cleans up output files after a successful run
+    """
+
+    def __init__(
+        self,
+        outgoing_directory: str,
+        pipeline_run_datetime: datetime,
+        delete_file_func: callable,
+        md5sum_func: callable,
+        stat_size_func: callable,
+        isdir_func: callable,
+        isfile_func: callable,
+        read_header_func: callable,
+        string_to_file_func: callable,
+        dry_run: bool = False,
+    ):
+        self.outgoing_directory = outgoing_directory
+        if not isdir_func(outgoing_directory):
+            raise ManifestError(
+                f"Outgoing directory does not exist: {self.outgoing_directory}"
+            )
+
+        if not isinstance(pipeline_run_datetime, datetime):
+            raise ManifestError("Pipeline run datetime must be a datetime object.")
+
+        self.manifest_datetime = pipeline_run_datetime.strftime("%Y%m%d_%H%M%S")
```

Comment:
Is it preferable to use "%Y-%m-%d %H:%M:%S.%f+00" format used in src/utils/hdfs_mods.py and src/utils/local_file_mods.py from recent 413 merge?